### PR TITLE
cleaned up your hash table, changed to vulndetails, added pagination and max records

### PR DESCRIPTION
**Pull request by AaronG1234**
## Description
1) your /rest/Analysis/ json body was all out of spec from the API, i threw out tons of unused keypairs
2) i changed it to return vulndetails (you may like summary data, perhaps another argument to flip between)
3) added ability to support multiple severities
4) added pagination and ability to specify maxrecords (just truncates when it hits)
5) added write-verbose to the pagination process

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by AaronG1234** [comment]
if the tests dont pass, the code is free to you :) im already using it....

**Comment by AaronG1234** [comment]
Will do
________________________________

**Comment by AaronG1234** [comment]
Ok, added help... worked on sort order (API is a mess) etc. etc.

File: PSTenable/Public/Get-PSTenableSeverity.ps1 [hunk 1]
```diff
@@ -4,88 +4,137 @@ function Get-PSTenableSeverity {
...
 
         $TokenExpiry = Invoke-PSTenableTokenStatus
         if ($TokenExpiry -eq $True) {Invoke-PSTenableTokenRenewal}
 
-        switch ($Severity) {
-            "Critical" { $ID = "4" }
-            "High" { $ID = "3" }
-            "Medium" { $ID = "2" }
-            "Low" { $ID = "1" }
+        if (!($Severity)) {
```
Comment (jwmoss) [inline PSTenable/Public/Get-PSTenableSeverity.ps1]: I'd prefer `if (-not $severity) {}` for readability for newcomers.

File: PSTenable/Public/Get-PSTenableSeverity.ps1 [hunk 2]
```diff
@@ -4,88 +4,137 @@ function Get-PSTenableSeverity {
...
+        Revision 0.1, Sept 2019, jwmoss
+        Revision 0.2, Nov 2019, aarong1234 
     .INPUTS
         None
     .PARAMETER Severity
-        Option for Critical, High, Medium or Low.
+        Option for any of "Critical", "High", "Medium", "Low", "Al"l, "All with Info". Defaults to "Critical","High".  All with Info will get ALL vuln data
+    .PARAMETER MaxRecords
+        Option for maximum records (rows of data) that should be requested (as a throttle), Default is 0 (all records) [note: sorted by score, descending]
+    .PARAMETER Detail
```
Comment (jwmoss) [inline PSTenable/Public/Get-PSTenableSeverity.ps1]: Should be Detailed?

File: PSTenable/Public/Get-PSTenableSeverity.ps1 [hunk 3]
```diff
@@ -4,88 +4,137 @@ function Get-PSTenableSeverity {
...
-                "startOffset"  = 0
-                "endOffset"    = 5000
-                "filters"      = [array]@{
-                    "id"           = "severity"
-                    "filterName"   = "severity"
-                    "operator"     = "="
-                    "type"         = "vuln"
-                    "ispredefined" = $true
-                    "value"        = "$ID"
+        $ID | % { #because of lack of useful sorting data (scores, time) within a severity, especially in summary data... we are going to query severities in order
```
Comment (jwmoss) [inline PSTenable/Public/Get-PSTenableSeverity.ps1]: Replace `%` with `Foreach-Object`

**Comment by AaronG1234** [comment]
ok, all comments merged

**Comment by jwmoss** [comment]
Sweet! I’ll merge into master sometime tonight or tomorrow.

**Pull request merged**